{{cookiecutter.repostory_name}}/app/Dockerfile

```diff
@@ -1,19 +1,38 @@
-FROM python:3.6
+{% if cookiecutter.use_alpine_linux == "y" -%}
+ARG BASE_IMAGE=python:3.6-alpine
+{% else -%}
+ARG BASE_IMAGE=python:3.6
+{% endif -%}
 
-COPY ./bin/s6-overlay-amd64.tar.gz /tmp/s6-overlay-amd64.tar.gz
+FROM $BASE_IMAGE AS base-image
 
-RUN tar xvfz /tmp/s6-overlay-amd64.tar.gz -C / \
-    && rm -f /tmp/s6-overlay-amd64.tar.gz {% if cookiecutter.use_celery == "y" %}\
-    && adduser --disabled-password --gecos '' celery{% endif %}
+WORKDIR /root/src/
+ENV PYTHONUNBUFFERED=1
+ENV PATH=/root/.local/bin:$PATH
 
-COPY ./src/ /srv/app/
+RUN pip3 install --user --no-cache-dir gunicorn==19.9.0
 
-WORKDIR /srv/app
+COPY ./src/requirements.txt requirements.txt
 
-RUN pip3 install --no-cache-dir -r requirements.txt gunicorn\
-    && python3 -m compileall -b .
+{% if cookiecutter.use_alpine_linux == "y" -%}
+RUN grep psycopg2 requirements.txt && \
+    apk add postgresql-dev gcc python3-dev musl-dev
+{% endif -%}
 
-ENTRYPOINT ["/init"]
+RUN pip3 install --user --no-cache-dir -r requirements.txt
+
+COPY ./src/ /root/src/
+
+RUN python3 -m compileall -b -f -q .
```

Comment:
I'm not sure if to compile just django project files or python dependencies installed in `/root/.local` as well?

Reply:
my understanding is that whatever is not compiled at that stage, will be compiled in runtime

Reply:
my understanding is that whatever is not compiled at that stage, will be compiled in runtime

Reply:
It will, right.
I was just thinking if there's a point in compiling all of the code of dependencies when sometimes you just use 10% of the library functionality. If a few of additional MB added to the image is fine we can compile everything.